java-checks/src/main/java/org/sonar/java/checks/security/CipherBlockChainingCheck.java

```diff
@@ -56,6 +70,24 @@ protected void onConstructorFound(NewClassTree newClassTree) {
     }
   }
 
+  private static boolean isDynamicallyGenerated(ExpressionTree tree) {
+    if (tree.is(Tree.Kind.IDENTIFIER)) {
+      Symbol symbol = ((IdentifierTree) tree).symbol();
+      VariableTree declaration = ((Symbol.VariableSymbol) symbol).declaration();
```

Comment:
`ECJ` can be quite tricky here. In case of strange resolution/recovery from its engine, it might assign an unknown symbol rather than a `VariableSymbol`. Before casting, I would check that the symbol is not unknown.

Reply:
Good to know!